### ISSUE TITLE
Feature request: Add support for sandboxing

### ISSUE THREAD

**Issue by pastoraleman** [bluejekyll/trust-dns#12]
It would be useful to have support for sandboxing trust-dns, in particular decoding of requests.

[rusty-sandbox](https://github.com/myfreeweb/rusty-sandbox) offers an abstraction layer over the sandboxing mechanisms provided by several operating systems, and could be incorporated as an optional compile-time feature.

**Comment by bluejekyll** [comment]
Looks like Linux support is lacking. I don't have a good way to test the other OSes right now. If I incorporate this is it something you could test? I really like the idea.

**Comment by pastoraleman** [comment]
Yes, happy to do some testing. I can test against OS X and FreeBSD, and can also test against OpenBSD when rusty-sandbox's pledge support is completed.

**Comment by phillipCouto** [comment]
On linux wouldn't the tools like selinux, systemd, and apparmor provide very similar features without the application being coded to support the environment?

Also there is no support for windows in rust-sandbox or the mentioned gaol library either. For completeness isn't this something to consider as well?

**Comment by bluejekyll** [comment]
I have done zero research on this. selinux, apparmor, even docker/lxc, would be fine methods of sandboxing the process.
